### ISSUE TITLE
CJK is not handled properly

### ISSUE THREAD

**Issue by tete1030** [Maximus5/cygwin-connector#18]
Hi,
Thank you for your great work. I found some problems about CJK when using cygwin-connector. Here are my tests.
I created two test console, one using connector and the other not using it, and commands of them are:
**with connector:**
> set CHERE_INVOKING=1 & E:\cygwin64\bin\conemu-cyg-64.exe E:\cygwin64\bin\bash.exe --login -i -new_console:C:"E:\cygwin64\Cygwin.ico"

**without connector:**
> set CHERE_INVOKING=1 & E:\cygwin64\bin\bash.exe --login -i -new_console:C:"E:\cygwin64\Cygwin.ico"

### `Compress long strings to fit space` is enabled
![1](https://cloud.githubusercontent.com/assets/4312115/19828404/668d679a-9df7-11e6-8c7a-b967e5b17bf1.png)
(Figure 1 and Figure 2)
### `Compress long strings to fit space` is disabled
![2](https://cloud.githubusercontent.com/assets/4312115/19828472/f926f6f6-9df8-11e6-95a3-2fdaa5fee705.png)
(Figure 3 and Figure 4)

The above test shows two problems. **First**, using of connector makes conemu or cygwin bash take width of a CJK character as only one space instead of two. You can see that in Figure 4 the "(Ssh Users)" string is properly wrapped to the next line while not in Figure 3. **Second**, when `Compress long strings to fit space` option is enabled, lines no need to be compressed is still compressed. This can be seen by compare Figure 2 against Figure 4. But this might be an issue of conemu not cyg-connector.

## Test 2
![3](https://cloud.githubusercontent.com/assets/4312115/19828590/d37d5a90-9dfc-11e6-82a2-8d2845aa55b9.png)
(Figure 5)

The cause of this should be same as Test 1.

## Test 3

1. Input a string with two Chinese characters followed.
![4](https://cloud.githubusercontent.com/assets/4312115/19828613/c1dcc478-9dfd-11e6-80e5-037c96ff2701.png)

2. Press backspace twice.
![5](https://cloud.githubusercontent.com/assets/4312115/19828619/ed82c352-9dfd-11e6-82ea-fe6be9b3cc91.png)

3. further press backspace 7 times.
![6](https://cloud.githubusercontent.com/assets/4312115/19828653/9f5be1da-9dfe-11e6-8229-1fefe2cfe2f9.png)

**Comment by jakitliang** [comment]
Yeah, it is the problem with ConEmu [ msys2 connector | cygwin connector ].

The connector Asia charater stick many charactor as one,
when you place your cursor form “|你好" and press 1 time,
your cursor position will goes to "你好  |",
which is a wrong behavior.

The expected behavior shoud be "你|好",
                                                      ^- Look at the cursor is here.

I tried those ways:

- chcp utf8
- set LANG...
- settings "mono font"
- change font type

Nothing works.
And Finally, I give up ConEmu and Cmder.

Such bad Emulators, I totally don't know why they choose.

Now 2023, time goes by, bugs never be fixed.
